apiserver/facades/client/application/deployrepository.go

```diff
@@ -630,78 +629,120 @@ func (v *deployFromRepositoryValidator) deducePlatform(arg params.DeployFromRepo
 		return corecharm.Platform{}, usedModelDefaultBase, err
 	}
 
-	// Match against platforms from placement
 	placementPlatform, placementsMatch, err := v.platformFromPlacement(arg.Placement)
-	if err != nil && !errors.Is(err, errors.NotFound) {
+	if err != nil {
 		return corecharm.Platform{}, usedModelDefaultBase, err
 	}
-	if err == nil && !placementsMatch {
-		return corecharm.Platform{}, usedModelDefaultBase, errors.BadRequestf("bases of existing placement machines do not match")
+	// No machine scoped placement to match, return after checking
+	// if using default model base.
+	if placementPlatform == nil {
+		return v.modelDefaultBase(platform)
+	}
+	// There can be only 1 platform.
+	if !placementsMatch {
+		return corecharm.Platform{}, usedModelDefaultBase, errors.BadRequestf("bases of existing placement machines do not match each other")
```

Comment:
Can we include the mismatched bases in the error

Reply:
Adding it here in the error is a bit messy.  Adding an Error log with the data instead as we can predict when it'll fail.